### ISSUE TITLE
Inaccurate data extraction

### ISSUE THREAD

**Issue by Paasu** [neuropoly/axondeepseg#347]
**Describe the bug**
I tried extracting the morphometric data on the manual masks to test the efficacy of the data compared to my manual measurements. I do understand that it was the manual masks and not the segmented images so it might be the reason the data differs significantly. The reason I haven't used the masks to train and the use the segmented images to analyze is because of #346. 
[Neuron Morphometrics.pdf](https://github.com/neuropoly/axondeepseg/files/5183710/Neuron.Morphometrics.pdf)

I only used a very small sample of 4 masks but there seems to be a significant difference in g-ratio, and myelin thickness with smaller but present differences in axon diameter. Is it possible because my masks aren't high quality enough?
[Images.zip](https://github.com/neuropoly/axondeepseg/files/5183715/Images.zip)

Feel free to let me know if you guys need more information as I do understand it might be a little unclear.

**Desktop (please complete the following information):**
 - OS: [OpenSUSE]
 - Version [15.1]

**Comment by jcohenadad** [comment]
@Paasu this is the names of your images:
![image](https://user-images.githubusercontent.com/2482071/92402198-2799bb80-f0fd-11ea-9ac1-2502ed072cd3.png)

and this is the content of the pdf:
![image](https://user-images.githubusercontent.com/2482071/92402224-2f596000-f0fd-11ea-9448-43e17306cc16.png)

it is impossible for us to know which sample number corresponds to which image.

so, what you need to do is: 
- send us: one manual mask
- send us: the exact syntax you typed to compute automatic morphometrics + results of automatic morphometrics
- send us your manual measures

then, we will be able to help you.

**Comment by Paasu** [comment]
@jcohenadad 
Sorry about that. Here contains what you requested:
[data.zip](https://github.com/neuropoly/axondeepseg/files/5184017/data.zip)

**Comment by jcohenadad** [comment]
the file `pixel_size_in_micrometer.txt` is missing. Also, do you have a screenshot or any recording of your manual measurements? e.g.: which axons did you measure and how did you measure them.

**Comment by Paasu** [comment]
[pixel_size_in_micrometer.txt](https://github.com/neuropoly/axondeepseg/files/5184058/pixel_size_in_micrometer.txt)
Woops, I thought I included it.

**Comment by jcohenadad** [comment]
@Paasu it is possible that the axon number you are referring to are not the same as the ones used for the automatic morphometrics. Thanks to this recent implementation (https://github.com/neuropoly/axondeepseg/pull/342), we are now able to display axon numbers that correspond to the output XLS file, via FSLeye's ADS plugin. 

Here are the axon numbers from your mask:
![Screen Shot 2020-09-08 at 2 48 45 PM](https://user-images.githubusercontent.com/2482071/92516069-a1f03b80-f1e2-11ea-9fe0-fe40f934d8e0.png)

And here is the output morphometrics file: [morphometrics.xlsx](https://github.com/neuropoly/axondeepseg/files/5190428/morphometrics.xlsx)

So, at this point, could you please point us to 2-3 examples of measures that you did manually, using the axon numbering as per my figure above, so we can be sure we compare the same axons.

**Comment by Paasu** [comment]
Here are my manual measurements specifying a couple of cells as labelled in your image above: [manual measurements.xlsx](https://github.com/neuropoly/axondeepseg/files/5193897/manual.measurements.xlsx)

Let me know if you need anything else like a more detailed explanation of the measurement protocol, etc.

**Comment by mariehbourget** [comment]
Hi @Paasu, it would help to have a more details about your measurement protocols (I assume the red lines in your example above are the manual measures for axon diameters, axon+myelin diameters and myelin thickness).

In AxonDeepSeg, the calculation of the diameter is based on the axon area of the mask. For each axons, the equivalent diameter is computed, which is the diameter of a circle with the same area as the axon.
Similarly, AxonDeepSeg calculates the area of each axon+myelin pairs on the mask and its equivalent diameter. The myelin thickness is then calculated from the difference between the axon+myelin equivalent diameter and the axon equivalent diameter.

**Comment by Paasu** [comment]
Here is a more detailed protocol on the method of analysis conducted on the TEM images: [Image Analysis Protocol.docx](https://github.com/neuropoly/axondeepseg/files/5199087/Image.Analysis.Protocol.docx)
 All of the measurements were done using ImageJ. If anything is unclear, please let me know and I'll clarify.

**Comment by mariehbourget** [comment]
@Paasu 
Thanks for the protocol. The differences between the manual and automatic measurements seem to come from the different methods used.

From your protocol, the manual measurements are taken from the shortest distance passing through the center of the axon, which is different from what is done in AxonDeepSeg.

For each axon on the mask, AxonDeepSeg computes the axon diameter based on the area of the axon (white area), and the cell diameter based on the area of the axon+myelin object (white+gray area). The diameters are the “equivalent diameter”, which is the diameter of a circle with the same area. The myelin thickness is then calculated from the difference between the axon+myelin equivalent diameter and the axon equivalent diameter.

In AxonDeepSeg, we recommend to use this method to ensure that a mean diameter is computed that is not bias in any specific direction, and also takes into account that axons do not have a perfectly cylindrical shape.

**Comment by Paasu** [comment]
@mariehbourget 

Thanks for the clarification!

It seems both of our methods are used to reduce bias within the results, so I would expect the results to be similar. In this specific image I found that after removing false positives, the axon diameter seems to be around 0.68 microns with the spreadsheet you have provided and my manual measurements show around 0.52 microns. I think I'll have to compare more images to form an opinion. I'll compare more images and let you guys know how it goes.

**Comment by Paasu** [comment]
Sorry for the delay! 
I had a look at 10 images.

There is a statistically significant difference between manual measurements and automatic measurements (p = 0.00034) and a lack of correlation (r square = 0.00029).

I think the difference is mainly due to elliptical nature of optic nerve neurons vs the circular nature of the neurons found in the brain. It'll be interesting to see if any method that can capture the ON neuron measurement is possible or perhaps the manual measurements we have are lacking.

Here is the excel sheet with more statistical information: [Analysis of images.xlsx](https://github.com/neuropoly/axondeepseg/files/5342235/Analysis.of.images.xlsx)

**Comment by mariehbourget** [comment]
Thank you for the follow-up @Paasu.
In the literature, we found that both the equivalent diameter and the short axis of an ellipse are used as a measurement for diameter, with no clear consensus on the preferred method.

We opened the issue #363 to implement a new feature to compute morphometrics using the short axis of a fitted ellipse. This method will better match your manual measurements protocol of minimal distance closest to the center.
